app/queries/cohort_analytics_query.rb

```diff
@@ -41,13 +38,24 @@ def patient_counts(cohort_start, cohort_end, report_start, report_end)
     controlled_patients = controlled(followed_up_patients)
     uncontrolled_patients = followed_up_patients - controlled_patients
 
+    registered_counts = registered_patients.group(:registration_facility_id).size.symbolize_keys
+    followed_up_counts = followed_up_patients.group(:registration_facility_id).size.symbolize_keys
+    defaulted_counts = registered_counts.merge(followed_up_counts) do |_, registered, followed_up|
+      registered - followed_up
+    end
+
+    controlled_counts = controlled_patients.group(:registration_facility_id).size.symbolize_keys
+    uncontrolled_counts = followed_up_counts.merge(controlled_counts) do |_, followed_up, controlled|
+      followed_up - controlled
+    end
+
     {
-      registered: registered_patients.size,
-      followed_up: followed_up_patients.size,
-      defaulted: registered_patients.size - followed_up_patients.size,
-      controlled: controlled_patients.size,
-      uncontrolled: uncontrolled_patients.size
-    }
+      registered: { total: registered_patients.size, **registered_counts },
+      followed_up: { total: followed_up_patients.size, **followed_up_counts },
+      defaulted: { total: registered_patients.size - followed_up_patients.size, **defaulted_counts },
+      controlled: { total: controlled_patients.size, **controlled_counts },
+      uncontrolled: { total: uncontrolled_patients.size, **uncontrolled_counts }
+    }.with_indifferent_access
```

Comment:
Consider removing the individual `symbolize_keys` and do a `deep_symbolize_keys` at the very end.

Reply:
I tried making this change, and remembered why it's this way. The splat (`**registered_counts`) requires the keys to be symbols, and I find the splat operator a lot more readable and scalable than merging hashes together.

Reply:
Fair enough.